libcontainer/cgroups/fs/cpuset.go

```diff
@@ -14,6 +15,42 @@ import (
 	"github.com/opencontainers/runc/libcontainer/configs"
 )
 
+var (
+	cpusetLock     sync.Mutex
+	cpusetNoPrefix bool
+	cpusetFastPath bool
+)
+
+func getCpusetFile(path string, name string) string {
+	cpusetLock.Lock()
+	defer cpusetLock.Unlock()
+
+	// Only the v1 cpuset cgroup is allowed to mount with noprefix.
+	// See kernel source: https://github.com/torvalds/linux/blob/2e1b3cc9d7f790145a80cb705b168f05dab65df2/kernel/cgroup/cgroup-v1.c#L1070
+	// Cpuset cannot be mounted with and without prefix simultaneously.
+	// Commonly used in Android environments.
+
+	if !cpusetFastPath {
+		err := unix.Access(filepath.Join(path, "cpuset."+name), unix.F_OK)
+		if err == nil {
+			// Use the fast path only if we can access one type of mount for cpuset already
+			cpusetFastPath = true
+		} else {
+			err = unix.Access(filepath.Join(path, name), unix.F_OK)
+			if err == nil {
+				cpusetNoPrefix = true
+				cpusetFastPath = true
+			}
+		}
+	}
+
+	if cpusetNoPrefix {
+		return name
+	}
+
+	return "cpuset." + name
```

Comment:
I'd prefer the fast path first, since it ends in return. E.g.

```go
func getCpusetFile(path string, name string) string {
	cpusetLock.Lock()
	defer cpusetLock.Unlock()
	
	if cpusetFastPath {
		if cpusetNoPrefix {
			return name
		}
		return "cpuset." + name
	}
	
	// Find out whether prefix is needed.
	err := unix.Access.....
```

Reply:
done